scripts/devtool/main.go

```diff
@@ -0,0 +1,221 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"os/exec"
+	"os/signal"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/jackc/pgx/v4"
+	"github.com/joho/godotenv"
+
+	// Load postgres driver
+	_ "github.com/jackc/pgx/v4/stdlib"
+)
+
+var (
+	red    = color.New(color.FgHiRed)
+	green  = color.New(color.FgHiGreen)
+	yellow = color.New(color.FgHiYellow)
+)
+
+var (
+	reset        = false
+	startAdmin   = true
+	startRuntime = true
+	startUI      = true
+)
+
+func init() {
+	flag.BoolVar(&reset, "reset", false, "reset db")
+	flag.BoolVar(&startAdmin, "admin", true, "start admin server")
+	flag.BoolVar(&startRuntime, "runtime", true, "start runtime server")
+	flag.BoolVar(&startUI, "ui", true, "start ui")
+}
+
+func main() {
+	flag.Parse()
+	// check go version
+	versionString := runtime.Version()
+	major, minor := parseVersion(versionString)
+
+	if major > 1 || (major == 1 && minor >= 20) {
+		panic("require go version greater than 1.20")
+	}
+
+	// check node version
+	nodeVersion, err := exec.Command("node", "--version").Output()
+	if err != nil {
+		panic("Error executing the 'node --version' command:" + err.Error())
+	}
+
+	versionString = strings.TrimPrefix(string(nodeVersion), "v")
+	major, _ = parseVersion(versionString)
+	if major < 18 {
+		panic("require Node.js version greater than 18")
+	}
+
+	// check docker version
+	cmd := exec.Command("docker-compose", "--version")
+	err = cmd.Run()
+
+	if err != nil {
+		panic("require docker-compose")
+	}
+
+	// observability - otel,zipkin,prometheus
+	cmd = exec.Command("docker-compose", "-f", "scripts/observability/docker-compose.yaml", "up", "--no-recreate")
+	err = cmd.Start()
+	if err != nil {
+		panic("could not start observability services")
+	}
+
+	if reset {
+		cmd = exec.Command("docker-compose", "-f", "admin/docker-compose.yml", "down", "--volumes")
+		err = cmd.Run()
+		if err != nil {
+			panic("could not stop db")
+		}
+		yellow.Println("DELETED EXISTING POSTGRES")
+	}
+
+	// postgres
+	cmd = exec.Command("docker-compose", "-f", "admin/docker-compose.yml", "up", "--no-recreate")
+	yellow.Println("STARTING POSTGRES")
+	err = cmd.Start()
+	if err != nil {
+		panic("could not start db")
+	}
+	waitDBUP()
+
+	c := make(chan os.Signal, 2)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-c
+		yellow.Println("stopping services")
+		time.Sleep(2 * time.Second)
+		os.Exit(1)
+	}()
```

Comment:
Instead of waiting 2 seconds, I would suggest waiting indefinitely until the sub-process have been killed (e.g. to prevent leaving hanging Docker containers)

Also, I don't see how this actually sends a cancellation signal to the sub-processes? You could consider using `graceful.WithCancelOnTerminate` and `errgroup.WithContext` for coordinating errors and cancellation instead. (That should also ensure that the whole process exits if any of the sub-processes exit.)